src/Http/Http.Abstractions/src/PathString.cs

```diff
@@ -39,7 +39,7 @@ public PathString(string? value)
         /// <summary>
         /// The unescaped path value
         /// </summary>
-        public string? Value
+        public string Value
```

Comment:
I think this is a better way to go with `PathString`. While you could initialize it with a null `value`, it's ordinarily never null.

Reply:
I'd leave this type with the original definition, but update `HasValue` to have `[MemberNotNullWhen(true, nameof(Value))]`.

Reply:
While I agree it's possible for `Value` to be null, in the most common use-cases that users would encounter a `PathString` this would never be null and makes the calling code more cumbersome. In our codebase, we've attempted to model the most common case rather than the exceptional. That said, @Tratcher would know more about how common of an occurence a `null` PathString is.

Reply:
PathString is just a wrapper type, it's likelyhood of being null is contextual. They type itself should say yes it can be null, and we'd override that in specific contexts like Request.Path which never should be null.